cli/internal/cache/cache_fs.go

```diff
@@ -35,33 +34,37 @@ func newFsCache(opts Opts, recorder analytics.Recorder, repoRoot turbopath.Absol
 
 // Fetch returns true if items are cached. It moves them into position as a side effect.
 func (f *fsCache) Fetch(anchor turbopath.AbsoluteSystemPath, hash string, _unusedOutputGlobs []string) (bool, []turbopath.AnchoredSystemPath, int, error) {
-	cachedFolder := f.cacheDirectory.UntypedJoin(hash)
+	cachePath := f.cacheDirectory.UntypedJoin(hash + ".tar.gz")
 
 	// If it's not in the cache bail now
-	if !cachedFolder.DirExists() {
+	if !cachePath.FileExists() {
 		f.logFetch(false, hash, 0)
 		return false, nil, 0, nil
 	}
 
-	// Otherwise, copy it into position
-	err := fs.RecursiveCopy(cachedFolder.ToStringDuringMigration(), anchor.ToStringDuringMigration())
-	if err != nil {
-		// TODO: what event to log here?
-		return false, nil, 0, fmt.Errorf("error moving artifact from cache into %v: %w", anchor, err)
+	cacheItem, openErr := cacheitem.Open(cachePath)
+	defer func() { _ = cacheItem.Close() }()
```

Comment:
`Close` only needs to happen if we didn't error on `Open`. Otherwise we're not even guaranteed to have a thing to close.

Reply:
`Close` is safe to call and helps protects us future mistakes. I'm intending to keep the immediate deferred close just as a pattern, and I also intend to inspect the close error here.